src/main/java/fi/aalto/cs/apluscourses/model/Points.java

```diff
@@ -103,4 +123,19 @@ private static void parseSubmissions(@NotNull JSONObject exerciseJson,
     submissions.put(exerciseId, submissionIds);
   }
 
+  @Nullable
+  private static Long parseSubmissionId(String submissionUrl) {
+    if (submissionUrl.isEmpty()) {
+      return null;
+    }
+    if (submissionUrl.endsWith("/")) {
+      submissionUrl = submissionUrl.substring(0, submissionUrl.length() - 1);
+    }
+    try {
+      return Long.parseLong(submissionUrl.substring(submissionUrl.lastIndexOf("/") + 1));
+    } catch (Exception e) {
+      return null;
```

Comment:
Do you know which (runtime?) exceptions can be thrown? Considering catching only those that are due to URLs with no ID, or whatever legal cases that should lead to that `null` return.